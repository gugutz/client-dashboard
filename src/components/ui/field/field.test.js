import React from 'react'
import renderer from 'react-test-renderer'
import Field from './index'

describe('<Field /> component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Field
          name="something"
          label="Some label"
          placeholder="Some placeholder"
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should support label prop', () => {
    const tree = renderer
      .create(
        <Field
          name="password"
          type="password"
          label={false}
          placeholder="Some placeholder"
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should support type prop', () => {
    const tree = renderer
      .create(
        <Field
          name="password"
          type="password"
          label="Some label"
          placeholder="Some placeholder"
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
