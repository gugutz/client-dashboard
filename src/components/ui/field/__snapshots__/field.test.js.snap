// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Field /> component should render correctly 1`] = `
<div
  className="field"
>
  <label>
    Some label
  </label>
  <div
    className="control"
  >
    <input
      className="input is-medium"
      placeholder="Some placeholder"
      type="text"
    />
  </div>
</div>
`;

exports[`<Field /> component should support label prop 1`] = `
<div
  className="field"
>
  <div
    className="control"
  >
    <input
      className="input is-medium"
      placeholder="Some placeholder"
      type="password"
    />
  </div>
</div>
`;

exports[`<Field /> component should support type prop 1`] = `
<div
  className="field"
>
  <label>
    Some label
  </label>
  <div
    className="control"
  >
    <input
      className="input is-medium"
      placeholder="Some placeholder"
      type="password"
    />
  </div>
</div>
`;
